src/Api/DateTimeResult.php

```diff
@@ -9,16 +13,69 @@ class DateTimeResult extends \DateTime implements \JsonSerializable
 {
     /**
      * Create a new DateTimeResult from a unix timestamp.
-     *
+     * The Unix epoch (or Unix time or POSIX time or Unix
+     * timestamp) is the number of seconds that have elapsed since
+     * January 1, 1970 (midnight UTC/GMT).
      * @param $unixTimestamp
      *
      * @return DateTimeResult
+     * @throws Exception
      */
     public static function fromEpoch($unixTimestamp)
     {
         return new self(gmdate('c', $unixTimestamp));
     }
 
+    /**
+     * @param $iso8601Timestamp
+     *
+     * @return DateTimeResult
+     */
+    public static function fromISO8601($iso8601Timestamp)
+    {
+        if (is_numeric($iso8601Timestamp) || !is_string($iso8601Timestamp)) {
+            throw new ParserException('Invalid timestamp value passed to DateTimeResult::fromISO8601');
+        }
+        return new DateTimeResult($iso8601Timestamp);
+    }
+
+    /**
+     * Create a new DateTimeResult from an unknown timestamp.
+     *
+     * @param $timestamp
+     *
+     * @return DateTimeResult
+     * @throws ParserException|Exception
+     */
+    public static function fromTimestamp($timestamp, $expectedFormat = null)
+    {
+            if (empty($timestamp) || !(is_string($timestamp) || is_int($timestamp))) {
+                throw new ParserException('Invalid timestamp value passed to DateTimeResult::fromTimestamp');
+            }
+            try {
+                if ($expectedFormat == 'iso8601') {
+                    try {
+                        return self::fromISO8601($timestamp);
+                    }
```

Comment:
Style nitpick: `catch` should be on the same line as the closing bracket, as per PSR-2 for [try-catch](https://www.php-fig.org/psr/psr-2/#56-try-catch).